src/components/Events/Events.js

```diff
@@ -6,10 +6,18 @@ import { EventsList, EventFilters, Loading } from '../';
 import { eventsAPI } from '../../api';
 import styles from './Events.sass';
 
+const odataFilterTypes = ['searchText', 'startDate'];
+const nonOdataFiltertypes = ['location', 'range'];
+
 function hasMoreEventsToLoad(currentPage, totalPages) {
   return currentPage + 1 < totalPages;
 }
 
+function buildFilterValues(filters, filterTypes) {
```

Comment:
Ideally the component wouldn't know anything about the shape of the request and just pass a filter object to the service to handle. This could also be extracted in a `transform`. Not a blocker for me.